src/app/auth/confirm-login/confirm-login.component.ts

```diff
@@ -58,9 +60,24 @@ export class ConfirmLoginComponent implements OnInit {
   completeLoginProcess(): void {
     this.authService.changeAuthState(AuthState.AwaitingAuthentication);
     this.phaseService.initializeCurrentRepository();
+    this.logger.info(`ConfirmLoginComponent: Current repo is ${this.phaseService.currentRepo}`);
     this.userService
       .createUserModel(this.username)
-      .pipe(flatMap(() => this.githubEventService.setLatestChangeEvent()))
+      .pipe(
+        flatMap(() => {
+          const currentRepo = this.phaseService.currentRepo;
+          if (Repo.isEmptyRepo(currentRepo)) {
+            return of(false);
+          }
+          return this.phaseService.isValidRepository(currentRepo);
+        }),
+        flatMap((isValidRepository) => {
+          if (!isValidRepository) {
+            return new Observable();
+          }
+          return this.githubEventService.setLatestChangeEvent();
+        })
```

Comment:
`flatMap` is deprecated. Please use `mergeMap`. See https://rxjs.dev/api/operators/flatMap